usage-examples/go/atlas-sdk-go/internal/config/appcontext.go

```diff
@@ -0,0 +1,362 @@
+		time.Since(cachedAppContextTime) < cacheTTL {
+		cached := cachedAppContext
+		cacheMutex.RUnlock()
+		return cached, nil
+	}
+	cacheMutex.RUnlock()
+
+	if !ValidateEnvironment(env) {
+		if strictValidation {
+			return nil, fmt.Errorf("invalid environment: %s", env)
+		}
+		log.Printf("Warning: Unexpected environment '%s' may cause issues", env)
+	}
+	// :state-remove-start: copy
+	// Special handling for internal-only test environment
+	if env == "test" {
+		log.Printf("Using test environment")
+
+		// If TEST_MOCK_CONFIG is set, use mock configuration
+		if os.Getenv("TEST_MOCK_CONFIG") == "true" {
+			mockConfig, mockSecrets, err := getTestConfiguration()
+			if err != nil {
+				return nil, errors.WithContext(err, "loading test configuration")
+			}
+
+			appCtx := &AppContext{
+				Environment: env,
+				Config:      mockConfig,
+				Secrets:     mockSecrets,
+			}
+
+			// Cache the result
+			cacheMutex.Lock()
+			cachedAppContext = appCtx
+			cachedAppContextTime = time.Now()
+			cacheMutex.Unlock()
+
+			return appCtx, nil
+		}
+	}
+	// :state-remove-end:
+	// Load environment files
+	envFiles := []string{
+		fmt.Sprintf(".env.%s", env),
+		".env",
+	}
+
+	loaded := false
+	for _, file := range envFiles {
+		if err := godotenv.Load(file); err == nil {
+			log.Printf("Loaded environment from %s", file)
+			loaded = true
+			break
+		}
+	}
+
+	if !loaded {
+		log.Printf("Warning: No environment files found, using system environment variables only")
+	}
+
+	// Get config path from env var or use default
+	configPath := os.Getenv(EnvConfigPath)
+	if configPath == "" {
+		configPath = fmt.Sprintf(DefaultConfigFormat, env)
+	}
+
+	log.Printf("Loading configuration for environment: %s", env)
+	log.Printf("Using config file: %s", configPath)
+
+	secrets, err := LoadSecrets()
+	if err != nil {
+		return nil, errors.WithContext(err, "loading secrets")
+	}
+
+	config, err := LoadConfig(configPath)
+	if err != nil {
+		return nil, errors.WithContext(err, "loading config")
+	}
+
+	if err = config.Validate(env); err != nil {
+		return nil, errors.WithContext(err, "validating config")
+	}
+
+	log.Printf("Configuration loaded successfully: env=%s, baseURL=%s, orgID=%s",
+		env, config.BaseURL, config.OrgID)
+
+	// Create and initialize the AppContext
+	appCtx := &AppContext{
+		Environment: env,
+		Config:      config,
+		Secrets:     secrets,
+	}
+
+	// Cache the result
+	cacheMutex.Lock()
+	cachedAppContext = appCtx
+	cachedAppContextTime = time.Now()
+	cacheMutex.Unlock()
+
+	return appCtx, nil
+}
+
+// LoadAppContextWithContext initializes application context with environment-specific configuration using a provided context for cancellation support.
+// If explicitEnv is provided, it overrides the APP_ENV environment variable
+// If strictValidation is true, invalid environments will return an error
+func LoadAppContextWithContext(ctx context.Context, explicitEnv string, strictValidation bool) (*AppContext, error) {
```

Comment:
This is a nit, but when I first saw the name `AppContext` I thought it was something to do with `context.Context`. And now I see a func named `LoadAppContextWithContext`. Could we rename this symbol & functions to something else, like `AppConfig` or `AppSettings`, where it's easier to understand the intent and the func name won't be quite so comical?

Reply:
well, nit is now moot, since this file is deleted 👍